test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -52,8 +52,9 @@ populate_worker(thread_context *tc)
         for (uint64_t i = 0; i < tc->key_count; ++i) {
             /* Start a txn. */
             tc->transaction.begin();
-            if (!tc->insert(cursor, coll.id, i)) {
-                /* We failed to insert, and our transaction was rolled back retry. */
+            if (tc->insert(cursor, coll.id, i)) {
```

Comment:
You removed `!`, is that on purpose ? Shouldn't we rollback if we fail the insertion ?

Reply:
Yeah because the boolean return is "rollback required" is that confusing? I think it's only confusing as it was original the other way round.

Reply:
My bad, I did not see this function has been modified too.